lib/saputils.pm

```diff
@@ -81,4 +83,87 @@ sub calculate_hana_topology {
     return \%topology;
 }
 
+=head2 check_hana_topology
+    check_hana_topology([input => $saphanasr_showAttr_format_script_output]);
+
+    Expect the output of saputills::calculate_hana_topology as input.
+    Uses calculate_hana_topology to get a hash of hashes, and then
+    checks the output to make sure that the cluster is working and ready.
+
+    The checks performed are:
+    - All node_states are online
+    - All sync_states are either SOK or PRIM
+
+=cut
+
+
+sub check_hana_topology {
+    my (%args) = @_;
+    croak "Missing mandatory 'input' argument" unless $args{input};
+    record_info("cmd_out", $args{input});
+    my $topology = $args{input};
+
+    my $all_online = 1;
+    my $prim_count = 0;
+    my $sok_count = 0;
+
+    foreach my $host (keys %$topology) {
+        # Check node_state
+        unless ($topology->{$host}->{node_state} eq 'online') {
+            $all_online = 0;
+            last;
+        }
+
+        # Check sync_state
+        if ($topology->{$host}->{sync_state} eq 'PRIM') {
+            $prim_count++;
+        } elsif ($topology->{$host}->{sync_state} eq 'SOK') {
+            $sok_count++;
+        }
+    }
+
+    # Final check for conditions
+    if ($all_online && $prim_count == 1 && $sok_count == (keys %$topology) - 1) {
+        return 1;
+    } else {
+        return 0;
+    }
```

Comment:
Suggestion for lines L126-L130:
```
return 1 if ($all_online && $prim_count == 1 && $sok_count == (keys %$topology) - 1);
return 0;
```

Reply:
changed to 

return ($all_online && $prim_count == 1 && $sok_count == (keys %$topology) - 1);